executor/insert_common.go

```diff
@@ -773,6 +774,11 @@ func setDatumAutoIDAndCast(ctx sessionctx.Context, d *types.Datum, id int64, col
 	if err == nil && d.GetInt64() < id {
 		// Auto ID is out of range, the truncated ID is possible to duplicate with an existing ID.
 		// To prevent updating unrelated rows in the REPLACE statement, it is better to throw an error.
```

Comment:
Now the comment is out-of-sync with the code, since `err == nil` here.
Please update the comment to clarify why we return nil in this case.

Reply:
thanks, done